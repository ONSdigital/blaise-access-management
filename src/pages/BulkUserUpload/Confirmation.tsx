import React, {ReactElement, useState} from "react";
import {useHistory} from "react-router-dom";
import {ONSButton, ONSPanel} from "blaise-design-system-react-components";

interface Props {
    validUsers: number
    uploadUsers: () => void
}

function Confirmation({validUsers, uploadUsers}: Props): ReactElement {
    const [formError, setFormError] = useState<string>("");
    const [confirm, setConfirm] = useState<boolean | null>(null);
    const [loading, setLoading] = useState<boolean>(false);
    const [message, setMessage] = useState<string>("");
    const history = useHistory();

    async function confirmOption() {
        if (confirm === null) {
            setFormError("Select an answer");
            return;
        }
        if (!confirm) {
            history.push("/");
            return;
        }

        setLoading(true);
        setMessage("");

        uploadUsers();
    }

    return (
        <>
            {
                message !== "" &&
                <ONSPanel status={message.includes("success") ? "success" : "error"}>
                    <p>{message}</p>
                </ONSPanel>
            }

            <form className="u-mt-s">
                {
                    formError === "" ?
                        confirmDeleteRadios(validUsers, setConfirm)
                        :
                        <ONSPanel status={"error"}>
                            <p className="panel__error">
                                <strong>{formError}</strong>
                            </p>
                            {confirmDeleteRadios(validUsers, setConfirm)}
                        </ONSPanel>
                }

                <br/>
                <ONSButton
                    label={"Continue"}
                    primary={true}
                    loading={loading}
                    id="confirm-continue"
                    onClick={() => confirmOption()}/>
                {!loading &&
                <ONSButton
                    label={"Cancel"}
                    primary={false}
                    id="cancel-overwrite"
                    onClick={() => history.push("/")}/>
                }
            </form>
        </>
    );
}

function confirmDeleteRadios(validUsers: number, setConfirm: (value: (((prevState: (boolean | null)) => (boolean | null)) | boolean | null)) => void) {
    return (
        <fieldset className="fieldset">
            <legend className="fieldset__legend">
            </legend>
            <div className="radios__items">

                <p className="radios__item">
                        <span className="radio">
                        <input
                            type="radio"
                            id="confirm-upload"
                            className="radio__input js-radio "
                            value="True"
                            name="confirm-upload"
                            aria-label="Yes"
                            onChange={() => setConfirm(true)}
                        />
                        <label className="radio__label " htmlFor="confirm-upload">
                            Yes, upload {validUsers} valid user{(validUsers > 1 && "s")}
                        </label>
                    </span></p>
                <br/>
                <p className="radios__item">
                        <span className="radio">
                        <input
                            type="radio"
                            id="cancel-upload"
                            className="radio__input js-radio "
                            value="False"
                            name="confirm-upload"
                            aria-label="No"
                            onChange={() => setConfirm(false)}
                        />
                        <label className="radio__label " htmlFor="cancel-upload">
                            No, do not upload any users
                        </label>
                    </span>
                </p>
            </div>
        </fieldset>
    );
}

export default Confirmation;
