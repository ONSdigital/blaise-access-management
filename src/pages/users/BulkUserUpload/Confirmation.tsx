import React, { ReactElement, useState } from "react";
import { useNavigate } from "react-router-dom";
import { ONSButton, ONSPanel } from "blaise-design-system-react-components";
import converter from "number-to-words";

interface Props {
    validUsers: number
    uploadUsers: () => void
}

function Confirmation({ validUsers, uploadUsers }: Props): ReactElement {
    const [formError, setFormError] = useState<string>("");
    const [confirm, setConfirm] = useState<boolean | null>(null);
    const [loading, setLoading] = useState<boolean>(false);
    const [message, setMessage] = useState<string>("");
    const navigate = useNavigate();

    async function confirmOption() {
        if (confirm === null) {
            setFormError("Select an answer");
            return;
        }
        if (!confirm) {
            navigate("/users");
            return;
        }

        setLoading(true);
        setMessage("");

        uploadUsers();
    }

    return (
        <>
            {
                message !== "" &&
                <ONSPanel status={message.includes("success") ? "success" : "error"}>
                    <p>{message}</p>
                </ONSPanel>
            }

            <form className="ons-u-mt-s">
                {
                    formError === "" ?
                        confirmDeleteRadios(validUsers, setConfirm)
                        :
                        <ONSPanel status={"error"}>
                            <p className="ons-panel__error">
                                <strong>{formError}</strong>
                            </p>
                            {confirmDeleteRadios(validUsers, setConfirm)}
                        </ONSPanel>
                }

                <br/>
                <ONSButton
                    label={"Continue"}
                    primary={true}
                    loading={loading}
                    id="confirm-continue"
                    onClick={() => confirmOption()}/>
                {!loading &&
                <ONSButton
                    label={"Cancel"}
                    primary={false}
                    id="cancel-overwrite"
                    onClick={() => navigate("/users")}/>
                }
            </form>
        </>
    );
}

function confirmDeleteRadios(validUsers: number, setConfirm: (value: (((prevState: (boolean | null)) => (boolean | null)) | boolean | null)) => void) {
    return (
        <fieldset className="ons-fieldset">
            <legend className="ons-fieldset__legend">
            </legend>
            <div className="ons-radios__items">

                <p className="ons-radios__item">
                        <span className="ons-radio">
                        <input
                            type="radio"
                            id="confirm-upload"
                            className="ons-radio__input ons-js-radio "
                            value="True"
                            name="confirm-upload"
                            aria-label="Yes"
                            onChange={() => setConfirm(true)}
                        />
                        <label className="ons-radio__label " htmlFor="confirm-upload">
                            Yes, upload {converter.toWords(validUsers)} valid user{(validUsers > 1 && "s")}
                        </label>
                    </span></p>
                <br/>
                <p className="ons-radios__item">
                        <span className="ons-radio">
                        <input
                            type="radio"
                            id="cancel-upload"
                            className="ons-radio__input ons-js-radio "
                            value="False"
                            name="confirm-upload"
                            aria-label="No"
                            onChange={() => setConfirm(false)}
                        />
                        <label className="ons-radio__label " htmlFor="cancel-upload">
                            No, do not upload any users
                        </label>
                    </span>
                </p>
            </div>
        </fieldset>
    );
}

export default Confirmation;
