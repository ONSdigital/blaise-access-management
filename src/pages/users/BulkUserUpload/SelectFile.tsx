import React, { ReactElement, useState } from "react";
import { Collapsible, ONSButton } from "blaise-design-system-react-components";
import CSVReader from "react-csv-reader";
import { ImportUser } from "../../../interfaces";
import { SelectFileProps } from "../../../interfaces/usersPage";

function SelectFile({ setUsersToUpload, movePageForward }: SelectFileProps): ReactElement {

    const [buttonLoading, setButtonLoading] = useState<boolean>(false);
    const [uploadData, setUploadData] = useState<ImportUser[]>([]);

    function uploadUsers() {
        setButtonLoading(true);
        setUsersToUpload(uploadData);
        setButtonLoading(false);

        movePageForward();
    }

    const papaparseOptions = {
        header: true,
        dynamicTyping: true,
        skipEmptyLines: true,
        transformHeader: (header: string) => header.toLowerCase().replace(/\W/g, "_")
    };

    return (
        <>
            <h1 className="ons-u-mb-l">Bulk user upload</h1>

            <label className="ons-label" htmlFor="react-csv-reader-input">Select CSV file
                <br />
                <span className="ons-label__description">File type accepted is .csv only</span>
            </label>
            <CSVReader
                cssInputClass="ons-input ons-input--text ons-input-type__input ons-input--upload"
                parserOptions={papaparseOptions}
                onFileLoaded={(data) => setUploadData(data)}
            />

            <br />
            <ONSButton label={"Upload"} primary={true} onClick={() => uploadUsers()} loading={buttonLoading} />

            <Collapsible title="What format should the bulk upload file be?">
                <>
                    <p>The user file should be a Comma-Separated Values file (CSV) with the headings <em>user, password
                        and role</em>.
                        A blank template is available to download below.
                    </p>

                    <div className="ons-download">

                        <div className="ons-download__content">
                            <h3 className="ons-u-fs-m ons-u-mt-no ons-u-mb-xs">
                                <a href={process.env.PUBLIC_URL + "/users.csv"} download="users.csv" type="text/csv">
                                    Bulk user upload template file<span className="ons-u-vh">,
                                        CSV document download, 48 Bytes
                                    </span></a>
                            </h3>
                            <span className="ons-u-fs-s ons-u-mb-xs ons-download__meta"
                                aria-hidden="true">Poster, CSV, 48 Bytes</span>
                            <p className="ons-download__excerpt">Blank CSV file to upload multiple users at once.</p>
                        </div>
                    </div>
                </>
            </Collapsible>
        </>
    );
}

export default SelectFile;
