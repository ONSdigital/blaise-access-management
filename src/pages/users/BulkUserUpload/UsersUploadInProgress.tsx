import { ONSLoadingPanel } from "blaise-design-system-react-components";
import React, {ReactElement} from "react";

const UsersUploadInProgress = (): ReactElement => {
    return (
        <>
            <h1 className="u-mb-l">Upload in progress</h1>
            <ONSLoadingPanel/>
        </>
    );
};

export default UsersUploadInProgress;
