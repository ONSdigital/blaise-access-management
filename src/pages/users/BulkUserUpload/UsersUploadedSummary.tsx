import React, {ReactElement, useEffect, useState} from "react";
import {ErrorBoundary, ONSButton, ONSPanel} from "blaise-design-system-react-components";
import {UploadedUser} from "../../../../Interfaces";
import {useHistory} from "react-router-dom";
import converter from "number-to-words";


interface Props {
    usersUploaded: UploadedUser[]
    numberOfValidUsers: number
}

function UsersUploadedSummary({usersUploaded, numberOfValidUsers}: Props): ReactElement {
    const [listError, setListError] = useState<string>("Loading ...");
    const [numberOfCreatedUsers, setNumberOfCreatedUsers] = useState<number>(0);
    const history = useHistory();

    useEffect(() => {
        let createdUsers = 0;
        usersUploaded.map((user: UploadedUser) => {
            if (user.created) {
                createdUsers = createdUsers + 1;
            }
        });
        setNumberOfCreatedUsers(createdUsers);
    }, [usersUploaded]);

    function failedToUploadUserTable() {
        return <>
            <h2 className="u-mt-xl">Users that were not created</h2>
            <ErrorBoundary errorMessageText={"Failed to load audit logs."}>
                {
                    usersUploaded && usersUploaded.length > 0
                        ?
                        <table id="batch-table" className="table">
                            <thead className="table__head u-mt-m">
                            <tr className="table__row">
                                <th scope="col" className="table__header ">
                                    <span>Username</span>
                                </th>
                                <th scope="col" className="table__header ">
                                    <span>Upload status</span>
                                </th>
                            </tr>
                            </thead>
                            <tbody className="table__body">
                            {
                                usersUploaded.map(({name, created}: UploadedUser, index: number) => {
                                    if (!created) {
                                        return (
                                            <tr className="table__row" key={name + index}
                                                data-testid={"batch-table-row"}>

                                                <td className="table__cell ">
                                                    {name}
                                                </td>
                                                <td className="table__cell ">
                                                    <span
                                                        className={`status status--${(created ? "success" : "error")}`}>
                                                        {
                                                            created
                                                                ? "User created"
                                                                : "User not created"
                                                        }
                                                    </span>
                                                </td>
                                            </tr>
                                        );
                                    }
                                })
                            }
                            </tbody>
                        </table>
                        :
                        <ONSPanel>{listError}</ONSPanel>
                }
            </ErrorBoundary>
        </>;
    }

    return (
        <>
            <h1 className="u-mb-l">Uploaded <em>{converter.toWords(numberOfCreatedUsers)} of {converter.toWords(numberOfValidUsers)}</em> user{(numberOfValidUsers > 1 && "s")} successfully
            </h1>

            {
                (
                    numberOfCreatedUsers !== numberOfValidUsers ?
                        <ONSPanel status={"error"}>
                            <p>
                                Not all users were created successfully, you can review which users were not created
                                successfully below.
                            </p>
                        </ONSPanel>
                        :
                        <ONSPanel>
                            <p>
                                Users will appear in the table on the homepage and should be accessible in Blaise
                            </p>
                        </ONSPanel>
                )
            }

            <br/>
            <ONSButton label={"Return to homepage"} primary={true} onClick={() => history.push("/")}/>
            {
                (
                    numberOfCreatedUsers !== numberOfValidUsers && failedToUploadUserTable()
                )
            }
        </>
    );
}

export default UsersUploadedSummary;
