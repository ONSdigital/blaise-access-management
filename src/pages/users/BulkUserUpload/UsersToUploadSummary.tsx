import React, {ReactElement, useEffect, useState} from "react";
import {ErrorBoundary, ONSPanel} from "blaise-design-system-react-components";
import {ImportUser} from "../../../../Interfaces";
import Confirmation from "./Confirmation";
import converter from "number-to-words";
import pino from "pino";

const logger = pino();

interface Props {
    statusDescriptionList: ImportUser[]
    uploadUsers: () => void
}

function UsersToUploadSummary({statusDescriptionList, uploadUsers}: Props): ReactElement {
    const [userList, setUserList] = useState<ImportUser[]>([]);
    const [listError, setListError] = useState<string>("Loading ...");
    const [noValidUsers, setNoValidUsers] = useState<number>(0);

    useEffect(() => {
        setupUserList().then(() => logger.info("SetupUserList is complete"));
    }, [statusDescriptionList]);

    async function setupUserList() {
        setListError("Loading ...");

        let noValid = 0;
        statusDescriptionList.map((user: ImportUser) => {
            if (user.valid) {
                noValid = noValid + 1;
            }
        });

        setNoValidUsers(noValid);
        statusDescriptionList.sort((a, b) => (a.valid ? 1 : 0) - (b.valid ? 1 : 0));

        if (statusDescriptionList.length === 0) {
            setListError("No users found to upload");
        }

        await setUserList(statusDescriptionList);
    }

    return (
        <>
            <h1 className="ons-u-mb-l">Bulk upload <em>{converter.toWords(noValidUsers)}</em> user{(noValidUsers > 1 && "s")}?</h1>
            <ONSPanel>
                <p>{noValidUsers} of {userList.length} users are valid and will be uploaded. <em>Invalid users will not be uploaded.
                </em> You can review any issues in the table below.</p>
            </ONSPanel>

            <Confirmation validUsers={noValidUsers} uploadUsers={uploadUsers}/>
            <h2 className="ons-u-mt-xl">Users to upload</h2>
            <ErrorBoundary errorMessageText={"Failed to load audit logs."}>
                {
                    userList && userList.length > 0
                        ?
                        <table id="batch-table" className="ons-table">
                            <thead className="ons-table__head ons-u-mt-m">
                            <tr className="ons-table__row">
                                <th scope="col" className="ons-table__header ">
                                    <span>Username</span>
                                </th>
                                <th scope="col" className="ons-table__header ">
                                    <span>Role</span>
                                </th>
                                <th scope="col" className="ons-table__header ">
                                    <span>User validity</span>
                                </th>
                            </tr>
                            </thead>
                            <tbody className="ons-table__body">
                            {
                                userList.map(({name, role, valid, warnings}: ImportUser, index: number) => {

                                    return (
                                        <tr className="ons-table__row" key={name + index}
                                            data-testid={"batch-table-row"}>

                                            <td className="ons-table__cell ">
                                                {name}
                                            </td>
                                            <td className="ons-table__cell ">
                                                {role}
                                            </td>
                                            <td className="ons-table__cell ">
                                                <span className={`ons-status ons-status--${(valid ? "success" : "error")}`}>
                                                    {
                                                        valid
                                                            ? "Valid User"
                                                            : warnings.map((message) => {
                                                                return (`${message}. `);
                                                            })
                                                    }
                                                </span>
                                            </td>

                                        </tr>
                                    );
                                })
                            }
                            </tbody>
                        </table>
                        :
                        <ONSPanel>{listError}</ONSPanel>
                }
            </ErrorBoundary>
        </>
    );
}

export default UsersToUploadSummary;
