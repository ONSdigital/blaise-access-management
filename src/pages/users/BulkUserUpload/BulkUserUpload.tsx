import React, { ReactElement, useState } from "react";
import { ImportUser, UploadedUser } from "../../../../interfaces";
import { NewUser } from "blaise-api-node-client";
import UsersToUploadSummary from "./UsersToUploadSummary";
import SelectFile from "./SelectFile";
import { addNewUser } from "../../../utilities/http";
import UsersUploadedSummary from "./UsersUploadedSummary";
import Breadcrumbs from "../../../Components/Breadcrumbs";
import UsersUploadInProgress from "./UsersUploadInProgress";

function BulkUserUpload(): ReactElement {

    const [usersToUpload, setUsersToUpload] = useState<ImportUser[]>([]);
    const [numberOfValidUsers, setNumberOfValidUsers] = useState<number>(0);
    const [usersUploaded, setUsersUploaded] = useState<UploadedUser[]>([]);

    document.body.className = "ons-js-enabled";

    const Pages = {
        Upload: "Upload",
        ToUploadSummary: "Summary",
        InProgress: "InProgress",
        UploadedSummary: "UploadedSummary"
    };

    function MovePageToSummary() {
        setCurrentPage(Pages.ToUploadSummary);
    }

    async function UploadUsers() {
        setCurrentPage(Pages.InProgress);

        const uploadedUsersList: UploadedUser[] = [];
        let numberToUpload = 0;

        for (const user of usersToUpload) {
            if (!user.valid) {
                continue;
            }
            numberToUpload = numberToUpload + 1;

            const newUser: NewUser = {
                defaultServerPark: "",
                serverParks: [],
                name: user.name,
                password: user.password,
                role: user.role
            };
            const created = await addNewUser(newUser);

            uploadedUsersList.push({ name: user.name, created: created });
        }

        setUsersUploaded(usersUploaded.concat(uploadedUsersList));
        setNumberOfValidUsers(numberToUpload);
        setCurrentPage(Pages.UploadedSummary);
    }

    const [currentPage, setCurrentPage] = useState<string>(Pages.Upload);

    function returnPage(): ReactElement {
        switch (currentPage) {
            case Pages.Upload:
                return <SelectFile setUsersToUpload={setUsersToUpload} movePageForward={MovePageToSummary} />;
            case Pages.ToUploadSummary:
                return <UsersToUploadSummary usersToImport={usersToUpload} uploadUsers={UploadUsers} />;
            case Pages.InProgress:
                return <UsersUploadInProgress />;
            case Pages.UploadedSummary:
                return <UsersUploadedSummary key={usersUploaded.toString()} usersUploaded={usersUploaded}
                    numberOfValidUsers={numberOfValidUsers} />;
            default:
                return <SelectFile setUsersToUpload={setUsersToUpload} movePageForward={MovePageToSummary} />;
        }
    }

    return (
        <>
            <Breadcrumbs BreadcrumbList={
                [
                    { link: "/", title: "Home" }, { link: "/users", title: "Manage users" }
                ]
            } />

            <main id="main-content" className="ons-page__main ons-u-mt-no">
                {returnPage()}
            </main>
        </>
    );
}

export default BulkUserUpload;
