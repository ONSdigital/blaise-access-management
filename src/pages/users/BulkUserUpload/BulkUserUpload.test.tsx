import React from "react";
import {render, waitFor, cleanup, screen, fireEvent} from "@testing-library/react";
import "@testing-library/jest-dom";
import {act} from "react-dom/test-utils";
import {createMemoryHistory} from "history";
import {Router} from "react-router";
import { ImportUser } from "../../../../Interfaces";
import BulkUserUpload from "./BulkUserUpload";

describe("Upload summary tests", () => {

    var userList: ImportUser[] = [
    {
        name:'Jamie',
        password:'pass',
        role:'BDSS',
        valid:true,
        warnings:[]
    },
    {
        name:'Rob',
        password:'pass2',
        role:'DST',
        valid:true,
        warnings:[]
    },    
    {
        name:'Jamie',
        password:'pass',
        role:'BDSS',
        valid:true,
        warnings:[]
    },       
    ]

    beforeAll(() => {        
        // @ts-ignore`
        jest.spyOn(BulkUserUpload.prototype, 'getusersToUpload').mockReturnValue(userList);
       
    });

    it("view users page matches Snapshot", async () => {
        //arrange
        const history = createMemoryHistory();
        const view = render(
            <Router history={history}>
                <BulkUserUpload />
            </Router>
        );

        // act
        await act(async () => {
            const uploadButton = view.getByTestId(`button`);
            fireEvent.click(uploadButton);
            await new Promise(process.nextTick);
        });

        // assert
        await waitFor(() => {
            expect(view).toMatchSnapshot();
        });

    });

/*     it("should render correctly", async () => {
        const history = createMemoryHistory();
        render(
            <Router history={history}>
                <Users currentUser={signedInUser}/>
            </Router>
        );

        await act(async () => {
            await new Promise(process.nextTick);
        });

        await waitFor(() => {
            expect(screen.getByText(/Manage users/i)).toBeDefined();
            expect(screen.getByText(/Create new user/i)).toBeDefined();
            expect(screen.getByText(/TestUser123/i)).toBeDefined();
            expect(screen.getByText(/SecondUser/i)).toBeDefined();
            expect(screen.queryByText(/Loading/i)).not.toBeInTheDocument();
        });
    }); */

    afterAll(() => {
        jest.clearAllMocks();
        cleanup();
    });
});

