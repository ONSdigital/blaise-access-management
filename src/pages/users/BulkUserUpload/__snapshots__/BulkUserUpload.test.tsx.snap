// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Upload summary tests view users page matches Snapshot 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body
    class="ons-js-enabled"
  >
    <div>
      <nav
        aria-label="Breadcrumb"
        class="ons-breadcrumb"
      >
        <ol
          class="ons-breadcrumb__items ons-u-fs-s"
        >
          <li
            class="ons-breadcrumb__item"
            id="breadcrumb-0"
          >
            <a
              class="ons-breadcrumb__link"
              href="/"
            >
              Home
            </a>
            <svg
              class="ons-svg-icon"
              focusable="false"
              viewBox="0 0 8 13"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M5.74,14.28l-.57-.56a.5.5,0,0,1,0-.71h0l5-5-5-5a.5.5,0,0,1,0-.71h0l.57-.56a.5.5,0,0,1,.71,0h0l5.93,5.93a.5.5,0,0,1,0,.7L6.45,14.28a.5.5,0,0,1-.71,0Z"
                transform="translate(-5.02 -1.59)"
              />
            </svg>
          </li>
          <li
            class="ons-breadcrumb__item"
            id="breadcrumb-1"
          >
            <a
              class="ons-breadcrumb__link"
              href="/users"
            >
              Manage users
            </a>
            <svg
              class="ons-svg-icon"
              focusable="false"
              viewBox="0 0 8 13"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M5.74,14.28l-.57-.56a.5.5,0,0,1,0-.71h0l5-5-5-5a.5.5,0,0,1,0-.71h0l.57-.56a.5.5,0,0,1,.71,0h0l5.93,5.93a.5.5,0,0,1,0,.7L6.45,14.28a.5.5,0,0,1-.71,0Z"
                transform="translate(-5.02 -1.59)"
              />
            </svg>
          </li>
        </ol>
      </nav>
      <main
        class="ons-page__main ons-u-mt-no"
        id="main-content"
      >
        <h1
          class="ons-u-mb-l"
        >
          Bulk upload 
          <em>
            zero
          </em>
           user
          ?
        </h1>
        <div
          class="ons-panel ons-panel--info ons-panel--no-title  ons-u-mt-m"
        >
          <div
            class="ons-panel__body "
          >
            <p>
              0
               of 
              0
               users are valid and will be uploaded. 
              <em>
                Invalid users will not be uploaded.
              </em>
               You can review any issues in the table below.
            </p>
          </div>
        </div>
        <form
          class="ons-u-mt-s"
        >
          <fieldset
            class="ons-fieldset"
          >
            <legend
              class="ons-fieldset__legend"
            />
            <div
              class="ons-radios__items"
            >
              <p
                class="ons-radios__item"
              >
                <span
                  class="ons-radio"
                >
                  <input
                    aria-label="Yes"
                    class="ons-radio__input ons-js-radio "
                    id="confirm-upload"
                    name="confirm-upload"
                    type="radio"
                    value="True"
                  />
                  <label
                    class="ons-radio__label "
                    for="confirm-upload"
                  >
                    Yes, upload 
                    zero
                     valid user
                  </label>
                </span>
              </p>
              <br />
              <p
                class="ons-radios__item"
              >
                <span
                  class="ons-radio"
                >
                  <input
                    aria-label="No"
                    class="ons-radio__input ons-js-radio "
                    id="cancel-upload"
                    name="confirm-upload"
                    type="radio"
                    value="False"
                  />
                  <label
                    class="ons-radio__label "
                    for="cancel-upload"
                  >
                    No, do not upload any users
                  </label>
                </span>
              </p>
            </div>
          </fieldset>
          <br />
          <button
            class="ons-btn "
            data-testid="button"
            id="confirm-continue"
            type="button"
          >
            <span
              class="ons-btn__inner"
            >
              Continue
            </span>
          </button>
          <button
            class="ons-btn  ons-btn--secondary"
            data-testid="button"
            id="cancel-overwrite"
            type="button"
          >
            <span
              class="ons-btn__inner"
            >
              Cancel
            </span>
          </button>
        </form>
        <h2
          class="ons-u-mt-xl"
        >
          Users to upload
        </h2>
        <div
          class="ons-panel ons-panel--info ons-panel--no-title  ons-u-mt-m"
        >
          <div
            class="ons-panel__body "
          >
            No users found to upload
          </div>
        </div>
      </main>
    </div>
  </body>,
  "container": <div>
    <nav
      aria-label="Breadcrumb"
      class="ons-breadcrumb"
    >
      <ol
        class="ons-breadcrumb__items ons-u-fs-s"
      >
        <li
          class="ons-breadcrumb__item"
          id="breadcrumb-0"
        >
          <a
            class="ons-breadcrumb__link"
            href="/"
          >
            Home
          </a>
          <svg
            class="ons-svg-icon"
            focusable="false"
            viewBox="0 0 8 13"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M5.74,14.28l-.57-.56a.5.5,0,0,1,0-.71h0l5-5-5-5a.5.5,0,0,1,0-.71h0l.57-.56a.5.5,0,0,1,.71,0h0l5.93,5.93a.5.5,0,0,1,0,.7L6.45,14.28a.5.5,0,0,1-.71,0Z"
              transform="translate(-5.02 -1.59)"
            />
          </svg>
        </li>
        <li
          class="ons-breadcrumb__item"
          id="breadcrumb-1"
        >
          <a
            class="ons-breadcrumb__link"
            href="/users"
          >
            Manage users
          </a>
          <svg
            class="ons-svg-icon"
            focusable="false"
            viewBox="0 0 8 13"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M5.74,14.28l-.57-.56a.5.5,0,0,1,0-.71h0l5-5-5-5a.5.5,0,0,1,0-.71h0l.57-.56a.5.5,0,0,1,.71,0h0l5.93,5.93a.5.5,0,0,1,0,.7L6.45,14.28a.5.5,0,0,1-.71,0Z"
              transform="translate(-5.02 -1.59)"
            />
          </svg>
        </li>
      </ol>
    </nav>
    <main
      class="ons-page__main ons-u-mt-no"
      id="main-content"
    >
      <h1
        class="ons-u-mb-l"
      >
        Bulk upload 
        <em>
          zero
        </em>
         user
        ?
      </h1>
      <div
        class="ons-panel ons-panel--info ons-panel--no-title  ons-u-mt-m"
      >
        <div
          class="ons-panel__body "
        >
          <p>
            0
             of 
            0
             users are valid and will be uploaded. 
            <em>
              Invalid users will not be uploaded.
            </em>
             You can review any issues in the table below.
          </p>
        </div>
      </div>
      <form
        class="ons-u-mt-s"
      >
        <fieldset
          class="ons-fieldset"
        >
          <legend
            class="ons-fieldset__legend"
          />
          <div
            class="ons-radios__items"
          >
            <p
              class="ons-radios__item"
            >
              <span
                class="ons-radio"
              >
                <input
                  aria-label="Yes"
                  class="ons-radio__input ons-js-radio "
                  id="confirm-upload"
                  name="confirm-upload"
                  type="radio"
                  value="True"
                />
                <label
                  class="ons-radio__label "
                  for="confirm-upload"
                >
                  Yes, upload 
                  zero
                   valid user
                </label>
              </span>
            </p>
            <br />
            <p
              class="ons-radios__item"
            >
              <span
                class="ons-radio"
              >
                <input
                  aria-label="No"
                  class="ons-radio__input ons-js-radio "
                  id="cancel-upload"
                  name="confirm-upload"
                  type="radio"
                  value="False"
                />
                <label
                  class="ons-radio__label "
                  for="cancel-upload"
                >
                  No, do not upload any users
                </label>
              </span>
            </p>
          </div>
        </fieldset>
        <br />
        <button
          class="ons-btn "
          data-testid="button"
          id="confirm-continue"
          type="button"
        >
          <span
            class="ons-btn__inner"
          >
            Continue
          </span>
        </button>
        <button
          class="ons-btn  ons-btn--secondary"
          data-testid="button"
          id="cancel-overwrite"
          type="button"
        >
          <span
            class="ons-btn__inner"
          >
            Cancel
          </span>
        </button>
      </form>
      <h2
        class="ons-u-mt-xl"
      >
        Users to upload
      </h2>
      <div
        class="ons-panel ons-panel--info ons-panel--no-title  ons-u-mt-m"
      >
        <div
          class="ons-panel__body "
        >
          No users found to upload
        </div>
      </div>
    </main>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
